jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -1027,7 +1037,35 @@ ecma_builtin_json_serialize_array (ecma_json_stringify_context_t *context_p, /**
   const bool has_gap = !ecma_compare_ecma_string_to_magic_id (context_p->gap_str_p, LIT_MAGIC_STRING__EMPTY);
 
   /* 6. */
-  uint32_t array_length = ((ecma_extended_object_t *) obj_p)->u.array.length;
+  uint32_t array_length;
+
+#if ENABLED (JERRY_ES2015)
```

Comment:
```c
#if PROXY
  if (is_proxy) {
   use jerry_get_object_length, and check for error.
  }
else {
#endif
  array_length = ((ecma_extended_object_t *) obj_p)->u.array.length;
#if PROXY
  }
#endif
```

Reply:
Didn't we decide in a different pull request that we should use the current form of guarding in these cases, instead of having an extra guard for the closing bracket? We should keep it consistent.

Reply:
Sure, I've forgotten about it.